Build fails with clang 7 : `-fno-exceptions` vs throw.

This is caused by bsf disabling exceptions, and simdpp usage of a throw.
```
../bsf/Source/Foundation/bsfUtility/ThirdParty/simdpp/core/aligned_allocator.h:84:13: error: cannot use 'throw' with exceptions disabled
            throw std::length_error("aligned_allocator<T,A>::allocate() - Integer overflow.");
```
so should we replace this throw with an abort() ?

--- a/Source/Foundation/bsfUtility/ThirdParty/simdpp/core/aligned_allocator.h
+++ b/Source/Foundation/bsfUtility/ThirdParty/simdpp/core/aligned_allocator.h
@@ -80,9 +80,12 @@ public:
             return nullptr;
         }
 
-        if (n > max_size()) {
-            throw std::length_error("aligned_allocator<T,A>::allocate() - Integer overflow.");
-        }
+		if (n > max_size()) {
+            //BSF: replaced throw with a simple abort, since in `-fno-exceptions` mode the compiler would put
+            // an abort in here anyway.
+            //throw std::length_error("aligned_allocator<T,A>::allocate() - Integer overflow.");
+            abort();
+		}
 
 
         /* We align the storage by adding @a alignment bytes and truncating the

